R/ordering.R

```diff
@@ -82,3 +82,90 @@ setMethod("ordering<-", "DatasetCatalog", function (x, value) {
     }
     return(x)
 })
+
+#' Move a variable to after another variable
+#'
+#' @param x the variable group that `after` is in
+#' @param value the variable or dataset subset you would like to move
+#' @param after the variable you want to precede `value`
+#' @return returns a variable group
+#' @examples
+#' \dontrun{
+#' ordering(ds)[['Demographics']] <- moveToAfter(ordering(ds)[['Grp A']], ds['age'], ds$educ)
+#' }
+moveToAfter <- function(x, value, after){
+    if (!inherits(after, "OrderGroup")) {
+        after <- urls(after)
+    }
+    if (!inherits(value, "OrderGroup")) {
+        value <- urls(value)
+    }
+    whi <- ifelse(inherits(after, "OrderGroup"), which(names(x) %in% name(after)), which(entities(x) %in% after))
+    if ((inherits(value, "OrderGroup") && !name(value) %in% names(x)) || (!inherits(value, "OrderGroup") && !value %in% entities(x))) entities(x) <- c(entities(x), value)
+    whi2 <- ifelse(inherits(value, "OrderGroup"), which(names(x) %in% name(value)), which(entities(x) %in% value))
+    entities(x) <- entities(x)[c(setdiff(1:whi, whi2), whi2, setdiff((whi+1):length(entities(x)), whi2))]
+    return(x)
+}
+
+#' Copy the order from one dataset to another.
+#'
+#' @param source the dataset you want to copy the order from
+#' @param target the dataset you want to copy the order to
+#' @return returns an object of class [`VariableOrder`] (which can be assigned
+#' to a dataset with [`ordering`])
+#' @examples
+#' \dontrun{
+#' ordering(ds) <- copyOrder(ds1, ds)
+#' }
+#' @export
+copyOrder <- function(source, target){
+    if (!is.dataset(source) | !is.dataset(target)) {
+        halt("Both source and target must be Crunch datasets.")
+    }
+
+    ord <- entities(ordering(source))
+    new_ord <- lapply(ord, copyOrderGroup,  source = source, target = target)
+
+    # TODO: copy other VariableOrder features?
```

Comment:
I was imagining that we would put the appropriate target dataset values in here (admittedly, "copy" is not the right word here). One benefit of making sure that `catalog_url` is filled in properly is that the `show` method would work on the output of `copyOrder` since it uses `catalog_url` to fetch aliases. This might be useful if a user wanted to see what order they were copying over before committing  the reordering.

Example output with and without assigning `catalog_url`
```
[crunch] > new_order <- copyOrder(ds, ds_alt)
[crunch] > new_order
(Hidden variable)
(Hidden variable)
[+] Group A
    (Hidden variable)
    (Hidden variable)
[crunch] > new_order@catalog_url <- "http://local.crunch.io:8080/api/datasets/9bee9d4e24ec49229f5ad07e4a76f7ec/variables/?relative=on" # target dataset URL
[crunch] > new_order
v5
v6
[+] Group A
    v4
    v3
[crunch] > 
```